src/oatpp/core/data/mapping/type/Type.hpp

```diff
@@ -556,7 +556,7 @@ class Type {
 template <class T, class Clazz>
 template<class Wrapper>
 Wrapper ObjectWrapper<T, Clazz>::cast() const {
-  if(!Wrapper::Class::getType()->extends(m_valueType)) {
+  if(!m_valueType->extends(Wrapper::Class::getType())) {
```

Comment:
`ObjectWrapper<T, Clazz>::cast()` - is an unsafe static cast which allows downcasts to child classes. Thats why we check if `Wrapper::Class::getType()->extends(m_valueType)`.

Reply:
I did try to use assignment copy and played around with that a little but could only seem to get raw static casts to work, which is what is being done in `StandardCollection::PolymorphicDispatcher` so maybe that is the only answer for now